### DISCUSSION TITLE
changing Immersed boundary method geom after restart point

### DISCUSSION THREAD

**Discussion by Danfellows97** [firemodels/fds#12771] [Q&A] [answered]
Hi everyone, 

I'm currently trying to implement structural changes within FDS after a restart point while using the immersed boundary method. However, I am getting a forrtl sever 147 error (trying to access memory that's not there) when I swap the old bingeom file with a new one.

Is it because the geometry points have moved from their original interpolated position? 

If so I am looking into re-interpolating the geometry at the restart point and wanted to know if anyone has any experience with this.

All the best, 

Dan

**Comment by drjfloyd** [comment]
You can't make any changes during a restart that change the dimensions of the arrays that are written to the restart file. You can do things like change the magnitude of VEL on SURF as that doesn't change the number of variables FDS is tracking.  You could not, for example, change a SURF from TMP_FRONT to 1D heat conduction as that would change the size of the arrays holding wall temperature data.

**Reply by Danfellows97** [reply]
Hi Dr J Floyd, 

Thank you for the quick reply! 

If that is the case, how can the authors in the study "FDS-2Abaqus scripted CFD-FEM coupling C++ managed automated python: additionally assessing two-way coupling effectiveness" able to remove OBST geometry at the restart points? 

All the best, 

Dan

**Reply by drjfloyd** [reply]
Read the study. This is explained in detail in their report.

**Comment by marcosvanella** [comment, marked as answer]
The boundary cut-faces that get read into FDS for geoms (similar to OBSTS) will depend not only on the triangulation you are sending in and boundary conditions you have prescribed for it, but also on how that triangulation intersects the grid planes in your FDS cartesian meshes. Select solid and outline for geometry in smokeview and you will see the cut-face polygons I'm talking about. When you move the geometry, even if you have the same triangulation you are going to end up with a different set of boundary cut-faces. This is probably why you are seeing this memory error.
We should be able to trap this issue and make a graceful stop by checking we have the correct size of the CFACE array and correct BC applied to each CFACE.

________________________________

**Reply by Danfellows97** [reply]
Thank you for the detailed answer!
